File: invext-api/src/main/java/com/invext/domain/services/ServiceRequestService.java
package com.invext.domain.services;

import java.util.Optional;

import org.springframework.data.util.Pair;

import com.invext.domain.dtos.CreateServiceRequestDto;
import com.invext.domain.entities.ServiceRequest;

public interface ServiceRequestService {
    ServiceRequest createServiceRequest(CreateServiceRequestDto dto);
    Pair<ServiceRequest, Optional<ServiceRequest>> finishServiceRequest(Long serviceRequestId);
}

File: invext-app/src/services/ServiceRequestService.js
import axios from "../axios";

export class ServiceRequestService {

    createServiceRequest(serviceRequestData) {
        return axios.post('/service-requests', serviceRequestData);
    }

    finishServiceRequest(serviceRequestId) {
        return axios.put(`/service-requests/${serviceRequestId}/finished`);
    }
}
